semgrep-core/src/engine/Match_rules.ml

```diff
@@ -739,20 +758,21 @@ and satisfies_metavar_pattern_condition env r mvar opt_xlang formula =
                     lazy_ast_and_errors
                     (lazy content)
                     (Some r'))
-          | Some _lang, _mval ->
-              (* THINK: fatal error instead? *)
-              logger#error
-                "rule %s: metavariable-pattern: the content of %s is not text"
-                env.rule_id mvar;
+          | Some _lang, mval ->
+              logger#debug
```

Comment:
why not an error for this one?

Reply:
Initially I thought it should be an error too, but it can happen easily without being an error, I saw this when running the changes on semgrep-rules. I added this comment:
```ocaml
(* This is not necessarily an error in the rule, e.g. you may be
* matching `$STRING + ...` and then add a metavariable-pattern on
* `$STRING`. This will only work when `$STRING` binds to some text
* but it can naturally bind to other string expressions. *)
```